oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/InMemoryReactiveClientRegistrationRepository.java

```diff
@@ -46,11 +45,7 @@
 	 */
 	public InMemoryReactiveClientRegistrationRepository(ClientRegistration... registrations) {
 		Assert.notEmpty(registrations, "registrations cannot be empty");
-		this.clientIdToClientRegistration = new ConcurrentReferenceHashMap<>();
-		for (ClientRegistration registration : registrations) {
-			Assert.notNull(registration, "registrations cannot contain null values");
-			this.clientIdToClientRegistration.put(registration.getRegistrationId(), registration);
-		}
+		this.clientIdToClientRegistration = createClientRegistrationIdToClientRegistration(Arrays.asList(registrations));
```

Comment:
Sorry it wasn't clear. I was suggesting to keep `createClientRegistrationIdToClientRegistration` a private static method. Then `InMemoryReactiveClientRegistrationRepository` would no longer use a `Map`. Instead, it would use an instance of `InMemoryClientRegistrationRepository`. For example:

```java
this.delegate = new InMemoryClientRegistrationRepository(registrations);
```

```java
@Override
public Mono<ClientRegistration> findByRegistrationId(String registrationId) {
	return Mono.justOrEmpty(this.delegate.findByRegistrationId(registrationId));
}
```

Reply:
Okay, what about now?

Reply:
Thanks! The last thing is ...Can you please clean up the commit message to follow the guidelines in [Contributing](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#format-commit-messages)? I'm not super strict about the 50 character subject, but I do prefer it to be as concise as possible. Something like:

```
Make InMemory*ClientRegistrationRepository Consistent

The previous builders with the list argument were inconsistent with their 
respective builders of var args.
```

Reply:
It's done, thank you very much!